### PR TITLE
#1040: [Typo] AsyncStatusIndicator (closes #1040)

### LINKED ISSUE #1040: [Typo] AsyncStatusIndicator
← #1034

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ascariandrea**
Closes #1040

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: src/AsyncStatusIndicator/AsyncStatusIndicator.tsx [hunk 1]
```diff
@@ -27,16 +52,16 @@ export const Props = {
  */
 @skinnable()
 @props(Props)
-export default class AsyncStatusIndicator extends React.PureComponent {
+export default class AsyncStatusIndicator extends React.PureComponent<AsyncStatusIndicatorProps, void> {
```
Comment (gabro) [inline src/AsyncStatusIndicator/AsyncStatusIndicator.tsx]: you don't need `void`. `React.PureComponent<AsyncStatusIndicatorProps>` works the same

File: src/AsyncStatusIndicator/AsyncStatusIndicator.tsx [hunk 2]
```diff
@@ -27,16 +52,16 @@ export const Props = {
  */
 @skinnable()
```
Comment (FrancescoCioria) [inline src/AsyncStatusIndicator/AsyncStatusIndicator.tsx]: I would remove @skinnable completely from brc as we've been doing in AlinityPRO and GDSM

File: src/AsyncStatusIndicator/AsyncStatusIndicator.d.ts [hunk 3]
```diff
@@ -1,20 +1,6 @@
 import { CSSProperties, PureComponent } from 'react';
```
Comment (FrancescoCioria) [inline src/AsyncStatusIndicator/AsyncStatusIndicator.d.ts]: this file should be completely removed: Typescript will automatically generate it

**Comment by FrancescoCioria** [comment]
@ascariandrea I further improved the build in #1039 to:
- transpile `.tsx?` with both `tsc` and `babel`
- generate `.d.ts` automatically for `.tsx?`
- keep using only `babel` for `.js`
- keep moving manually written `.d.ts` from `src` to `lib`

now we can completely remove `.d.ts` and write `.tsx?` files as we would if we didn't have `babel` at all.
I rebased it on `1035-...` and added some commits.

NOTE: the config `declaration: true` requires `allowJs` to be `false` -> I had to refactor `utils` in TypeScript to make it work.

**Comment by gabro** [comment]
Since VSCode will look for docs on the props, can we also move them inline?

I mean

```ts
type Props = {
  /** the class name */
  className?: string
}

/** The Foo */
export default class Foo extends React.Component<Props>
```

instead of 

```ts
/** The Foo
  * @param className the class name
  */
export default class Foo extends React.Component<Props>
```

File: src/AsyncStatusIndicator/AsyncStatusIndicator.tsx (hunk 1 above)
Comment (giogonzo) [inline src/AsyncStatusIndicator/AsyncStatusIndicator.tsx]: also don't use void, it should be `State extends {}`.
`void` used to break in weird ways with `@types/react@15.0.38`

Comment (giogonzo) [inline src/AsyncStatusIndicator/AsyncStatusIndicator.tsx]: (probably not anymore in 16 :D https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L269)

**Comment by gabro** [comment]
rebased

**Pull request merged**